cmd/jaeger/internal/extension/jaegerquery/factory_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerquery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/jaegertracing/jaeger/ports"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/extension"
+)
+
+func Test_NewFactory(t *testing.T) {
+	factory := NewFactory()
+	assert.Equal(t, factory.Type(), component.Type("jaeger_query"))
+	assert.Equal(t, factory.CreateDefaultConfig(), createDefaultConfig())
+}
+
+func Test_CreateDefaultConfig(t *testing.T) {
+	expectedDefaultConfig := &Config{
```

Comment:
I don't like this - if default config keeps growing in the future, we'd need to keep duplicating it here. What benefit would this test provide?

Reply:
Replaced the config with a non-empty check to make sure that the default config is not an empty object.